integration_tests/src/main/python/cast_test.py

```diff
@@ -167,21 +158,27 @@ def test_cast_long_to_decimal_overflow():
         conf={'spark.sql.legacy.allowNegativeScaleOfDecimal': True})
 
 # casting these types to string should be passed
-basic_gens_for_cast_to_string = [byte_gen, short_gen, int_gen, long_gen, string_gen, boolean_gen, date_gen, null_gen, timestamp_gen] + decimal_gens_no_neg
+basic_gens_for_cast_to_string = [ByteGen, ShortGen, IntegerGen, LongGen, StringGen, BooleanGen, DateGen, TimestampGen] 
```

Comment:
Why exactly what this change made? It feels a little confusing to have the classes in here, when the other similarly named variables have the instances of the gen in them. It also is not clear what value this is providing over having just the gens in here.

Reply:
Because it is not easy to build the `data_gens` for `Map` from the provided instances of gens as the `_key_gen`   should be created with the declaration `nullable = False`.  
```python
basic_array_struct_gens_for_cast_to_string = [f() for f in basic_gens_for_cast_to_string] + [null_gen] + decimal_gens_no_neg
basic_map_gens_for_cast_to_string = [
    MapGen(f(nullable=False), f()) for f in basic_gens_for_cast_to_string] + [
    MapGen(DecimalGen(nullable=False), DecimalGen(precision=7, scale=3)), MapGen(DecimalGen(precision=7, scale=7, nullable=False), DecimalGen(precision=12, scale=2))]
```